cmd/mdatagen/internal/metadata.go

```diff
@@ -306,8 +318,31 @@ type Attribute struct {
 	FullName AttributeName `mapstructure:"-"`
 	// Warnings that will be shown to user under specified conditions.
 	Warnings Warnings `mapstructure:"warnings"`
-	// Optional defines whether the attribute is required.
-	Optional bool `mapstructure:"optional"`
```

Comment:
Should this be deprecated instead and stay here for a while?

Also do we know if this is used by components i.e. in Contrib? I couldn't find anything with a quick [search](https://github.com/search?q=repo%3Aopen-telemetry%2Fopentelemetry-collector-contrib+optional%3A+language%3AMarkdown&type=code&l=Markdown&p=1).

In any case this change might break downstream users of `mdatagen`, right? I'm not sure what is the adoption of `mdatagen` but I'd suggest deprecating this and warn users that still use so as to give them time to migrate if needed.